### PR TITLE
Update Package to Support PHP 8.0+ (v1.2)

### PR DESCRIPTION
**Pull request by justinyost**
Remove support for PHP 5/7. 

Updated CircleCI builds to install, test and code-sniff correctly on PHP 8.  

Updated all sub-classes for updated CodeSniffs, removes deprecation warnings for PHP 8, updated PHPUnit support to PHPUnit 9.0. 

Post this PR being merged, we're going to increment the version number to 1.2.

### CODE REVIEW

File: src/com/amazon/paapi5/v1/BrowseNode.php [hunk 1]
```diff
@@ -401,7 +401,7 @@ public function setSalesRank($salesRank)
      *
      * @return boolean
      */
-    public function offsetExists($offset)
+    public function offsetExists(mixed $offset): bool
```
Comment (lisavogtsf) [inline src/com/amazon/paapi5/v1/BrowseNode.php:404]: Was this type hinting done as part of the code sniffing you mentioned?

**Review by lisavogtsf — COMMENTED** [review]
Do we worry at all about staying in sync with Amazon? This feels like a lot of changes to something that is fundamentally Amazon's code.

Also is it expected that all the tests are 'Risky' or `This test did not perform any assertions` in local? Or am I possibly running them wrong?

File: composer.json [hunk 2]
```diff
@@ -28,21 +28,21 @@
         }
     ],
     "require": {
-        "php": ">=5.5",
+        "php": "^8.0",
```
Comment (enricobono) [inline composer.json:31]: Is this upgrade considered a _breaking change_ (thus we need to bump version to 2.0.0)?

Comment (justinyost) [reply composer.json:31]: @enricobono I don't think so based upon my understanding, referring to this post that sums up how I'm thinking about it https://medium.com/@sampart/semantic-versioning-when-you-change-the-required-programming-language-version-16a3a3555c95. 

The short version is: We're not changing the actual API for the package in any breaking way, merely the required PHP version and Composer will handle that if you don't have PHP 8 available on your machine you'll get the last version that does work for you.

File: src/com/amazon/paapi5/v1/BrowseNode.php (hunk 1 above)
Comment (justinyost) [reply src/com/amazon/paapi5/v1/BrowseNode.php:404]: Yep, we needed to add the typehints for the various [ArrayAccess](https://www.php.net/manual/en/class.arrayaccess.php) methods as they were presenting a Deprecation Warning before this.

**Comment by justinyost** [comment]
@lisavogtsf

I don't necessarily want to be out of synch with Amazon's code, but also we need code that works for us and is easily installable. I'll include some documentation to note more explicitly that this package is not a 1:1 copy of Amazon's code in the Readme. 

Yes, that is indeed what the tests are reporting, I think it's a place we should look at making some improvements again beyond what Amazon have done, but not really in scope for our current work.

**Comment by justinyost** [comment]
@lisavogtsf Commit that adds explicit notes about what's changed from Amazon's original code: https://github.com/thewirecutter/paapi5-php-sdk/pull/13/commits/5ab2c8f87d71d7d927d9003c2b460a1f9c4de241

**Review by lisavogtsf — APPROVED** [review]
Thank you for the clarifications, this looks good to me, and I agree on the minor version change.

File: composer.json (hunk 2 above)
Comment (enricobono) [reply composer.json:31]: Make sense, thanks.

**Review by enricobono — APPROVED** [review]

**Pull request merged**